lib/modules/dosomething/dosomething_helpers/dosomething_helpers.module

```diff
@@ -73,6 +73,55 @@ function dosomething_helpers_convert_date($date, $format = 'ISO-8601') {
 }
 
 /**
+ * Convert set of nested array $filters into one dimensional $parameters array.
+ * Needed for ApiCache Class ID generation.
+ *
+ * @param  array  $filters
+ * @return null|array
+ */
+function dosomething_helpers_convert_filters_to_url_parameters(array $filters = []) {
+  if (!$filters) {
+    return NULL;
+  }
+
+  $parameters = [];
+
+  foreach ($filters as $key => $value) {
+    if (!is_int($value) && !$value) {
+      continue;
+    }
+
+    if (!is_array($value)) {
+      $value = [$value];
+    }
+
+    switch($key) {
+      case 'nid':
+            $parameters['campaigns'] = implode(',', $value);
+            break;
+
+      case 'exclude':
+            $parameters[$key] = implode(',', $value);
+            break;
+
+      case 'load_user':
+            $parameters[$key] = 'true';
+            break;
+
+      case 'random':
+            $parameters[$key] = 'true';
+            break;
```

Comment:
Oh duh, I didn't realize you're doing the pass through in the switch in the example! cooooool! very smart. I'll give it a whirl!